pylops_mpi/basicoperators/BlockDiag.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import scipy as sp
+from mpi4py import MPI
+from typing import Optional, Sequence
+
+from pylops.utils import DTypeLike
+from pylops import LinearOperator, MatrixMult
+
+from pylops_mpi.DistributedArray import local_split, Partition
+
+sp_version = sp.__version__.split(".")
```

Comment:
Since we are building a new library, I suggest we do not support early versions of spicy (like we need to do for backward compatibility in pylops). Simply use `from scipy.sparse.linalg._interface import _get_dtype` and make 1.8.0 the minimum requirement

Reply:
Updated scipy to `1.8.0`, python is also now `3.8`(as required by scipy), updated the code, also updated `requirements-dev.txt`, `environment.yml` and `environment-dev.yml`. Also removed `3.7` from build.yaml.

Reply:
Perfect, I don't think we need to maintain python3.7 given that pretty much everyone uses 3.8+ these days...